google/auth/_default.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Application default credentials.
+
+Implementes application default credentials and project ID detection."""
+
+import io
+import json
+import os
+
+from six.moves import configparser
+
+from google.auth import compute_engine
+from google.auth import exceptions
+from google.auth.compute_engine import _metadata
+import google.auth.transport._http_client
+from google.oauth2 import service_account
+import google.oauth2.credentials
+
+# Environment variable for explicit application default credentials and project
+# ID.
+_CREDENTIALS_ENV = 'GOOGLE_APPLICATION_CREDENTIALS'
+_PROJECT_ENV = 'GCLOUD_PROJECT'
```

Comment:
1. You mentioned this a few times, would you like me to move all cloud sdk related stuff to its own module? I could get behind that.
2. This environment variable originated in `google-cloud-python` as far as I know, it has no bearing on the sdk.